File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.cpp
class Solution {
public:
    int minimumPushes(string s) {
        vector<int> v(26,0);
        for(int i =0;i<s.size();i++){
            v[s[i]-'a']++;
        }
        sort(v.rbegin(),v.rend());
        int c =8,ans=0;
        for(int i=0;i<26;i++){
            ans += v[i]*(c++/8);
        }
        return ans;

    }
};

File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.java
class Solution {
    public int minimumPushes(String s) {
        int[] fr = new int[26];

for (int i = 0; i < s.length(); i++) {
    fr[s.charAt(i) - 'a']++;
}

Arrays.sort(fr);

int c = 8, ans = 0;

for (int i = 25; i >= 0; i--) {
    ans += fr[i] * (c++ / 8);
}

return ans;

    }
}